syftlib/src/main/java/org/openmined/syft/Syft.kt

```diff
@@ -8,19 +8,44 @@ import org.openmined.syft.network.NetworkMessage
 import org.openmined.syft.network.SignallingClient
 import org.openmined.syft.threading.ProcessSchedulers
 
-class Syft(
+class Syft private constructor(
     private val workerId: String,
     private val keepAliveTimeout: Int = 20000,
     private val url: String,
     private val schedulers: ProcessSchedulers
 ) {
-
     private lateinit var signallingClient: SignallingClient
     private val compositeDisposable = CompositeDisposable()
 
+    companion object {
+        operator fun invoke(
+            workerId: String,
+            keepAliveTimeout: Int = 20000,
+            url: String,
+            schedulers: ProcessSchedulers
+        ): Syft? =
+                Syft(workerId, keepAliveTimeout, url, schedulers).takeIf { isValid(it) }
+
+        private fun isValid(syft: Syft): Boolean {
+            return checkProtocol(syft.url) &&
+                   checkKeepAliveTimeout(syft.keepAliveTimeout)
+        }
+
+        private fun checkProtocol(url: String) =
```

Comment:
what do you think, Rather than checking the protocol is wss what if construct the url in that way?  A set of constants for accepted protocols (currently wss)? There would be no need for the user to face this null error, I believe

Reply:
From a client perspective I would expect them to use a complete url including the protocol.
It is also easier to remove a check than to start building the different possibilities that might be in the future.

We are telling the user (probably a developer) that something was wrong and writing a message about it.